src/container/mod.rs

```diff
@@ -0,0 +1,226 @@
+use egui::Rect;
+
+use crate::Tree;
+
+use super::{Behavior, DropContext, SimplifyAction, TileId, Tiles};
+
+mod grid;
+mod linear;
+mod tabs;
+
+pub use grid::{Grid, GridLoc};
+pub use linear::{Linear, LinearDir, Shares};
+pub use tabs::Tabs;
+
+// ----------------------------------------------------------------------------
+
+/// The layout of a [`Container`].
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub enum Layout {
```

Comment:
irks me a bit that there is a dedicated enum that describes the different states of another enum (`Container`). Don't have a better solution to offer though